OpenSim/Common/Component.h

```diff
@@ -2487,11 +2501,18 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
     // subsystem.
     SimTK::ResetOnCopy<SimTK::MeasureIndex> _simTKcomponentIndex;
 
+    // list of subcomponents that are contained in this Component's properties
+    SimTK::Array_<SimTK::ReferencePtr<Component> >  _propertySubcomponents;
     // Keep fixed list of data member Components upon construction
     SimTK::Array_<SimTK::ClonePtr<Component> > _memberSubcomponents;
     // Hold onto adopted components
     SimTK::Array_<SimTK::ClonePtr<Component> > _adoptedSubcomponents;
 
+    // A flat list of all the subcomponents that comprise this Component.
+    // The list must be complete and in the correct System order by 
+    // addToSystem().
+    mutable std::vector<SimTK::ReferencePtr<const Component> > _orderedSubcomponents;
```

Comment:
Sorry; forget this comment. I need to get a better understanding of the changes first.